server/plugin/api.go

```diff
@@ -369,6 +401,13 @@ func (p *Plugin) completeConnectUserToGitHub(c *Context, w http.ResponseWriter,
 		MM34646ResetTokenDone: true,
 	}
 
+	appErr = p.API.KVDelete(stateToken)
+	if appErr != nil {
+		p.API.LogWarn("Failed to delete state token", "error", appErr.Error())
+		http.Error(w, "error deleting stored state", http.StatusBadRequest)
+		return
+	}
```

Comment:
This should be deleted from the KV store even if the OAuth process isn't successful, so I don't think it should move from where it originally was actually

Reply:
@mickmister keeping the code where it was is simply we are deleting it from KVStore with out even verifying the state and code Passed , 
i think if the state is some thing which passed is a uuid , once the oauth/complete API is access the previous state gets  
replaced , if we do without verify the details then we might loose the github user info and as well as sensitive information

Reply:
We should first verify that `state` is the correct format that we would expect to receive in general, whether it matches what we have in the KV store. If it's incorrect, return. If it's the correct format, we should delete from the KV store before moving on, because this is supposed to be a one-time process.

Reply:
sure